src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.LocalFunctionReferenceRewriter.cs

```diff
@@ -0,0 +1,146 @@
+                    return visited;
+                }
+
+                return (BoundCall)visited;
+            }
+
+            public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationExpression node)
+            {
+                if (node.MethodOpt?.MethodKind == MethodKind.LocalFunction)
+                {
+                    BoundExpression receiver;
+                    MethodSymbol method;
+                    var arguments = default(ImmutableArray<BoundExpression>);
+                    _lambdaRewriter.RemapLocalFunction(
+                        node.Syntax, node.MethodOpt, out receiver, out method, ref arguments);
+
+                    return new BoundDelegateCreationExpression(
+                        node.Syntax, receiver, method, isExtensionMethod: false, type: node.Type);
+                }
+
+                return base.VisitDelegateCreationExpression(node);
+            }
+
+            public override BoundNode VisitConversion(BoundConversion conversion)
+            {
+                if (conversion.ConversionKind == ConversionKind.MethodGroup &&
+                    conversion.SymbolOpt?.MethodKind == MethodKind.LocalFunction)
+                {
+                    BoundExpression receiver;
+                    MethodSymbol method;
+                    var arguments = default(ImmutableArray<BoundExpression>);
+                    _lambdaRewriter.RemapLocalFunction(
+                        conversion.Syntax, conversion.SymbolOpt, out receiver, out method, ref arguments);
+
+                    return new BoundDelegateCreationExpression(
+                        conversion.Syntax, receiver, method, isExtensionMethod: false, type: conversion.Type);
+                }
+                return base.VisitConversion(conversion);
+            }
+
+            public override BoundNode VisitBlock(BoundBlock node)
+            {
+                var newStatements = ArrayBuilder<BoundStatement>.GetInstance();
+
+                foreach (var statement in node.Statements)
+                {
+                    var replacement = (BoundStatement)this.Visit(statement);
+                    if (replacement != null)
+                    {
+                        newStatements.Add(replacement);
+                    }
+                }
+
+                return node.Update(node.Locals, node.LocalFunctions, newStatements.ToImmutableAndFree());
+            }
+        }
+
+
+        private void RemapLocalFunction(
+            CSharpSyntaxNode syntax,
+            MethodSymbol symbol,
+            out BoundExpression receiver,
+            out MethodSymbol method,
+            ref ImmutableArray<BoundExpression> parameters,
+            ImmutableArray<TypeSymbol> typeArguments = default(ImmutableArray<TypeSymbol>))
+        {
+            Debug.Assert(symbol.MethodKind == MethodKind.LocalFunction);
+
+            var constructed = symbol as ConstructedMethodSymbol;
+            if (constructed != null)
```

Comment:
>constructed != null [](start = 16, length = 19)

Could this test instead be `if (symbol != (object)symbol.ConstructedFrom)` ? I prefer to avoid depending on the precise hierarchy of implementation types.